src/test/java/teammates/sqlui/webapi/GetNotificationsActionTest.java

```diff
@@ -0,0 +1,310 @@
+    }
+
+    @Test
+    public void testExecute_invalidUserType_shouldFail() {
+        loginAsInstructor(GOOGLE_ID);
+
+        // when usertype is GENERAL
+        verifyHttpParameterFailure(Const.ParamsNames.NOTIFICATION_TARGET_USER,
+                NotificationTargetUser.GENERAL.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL,
+                String.valueOf(true));
+
+        // when usertype is a random string
+        verifyHttpParameterFailure(Const.ParamsNames.NOTIFICATION_TARGET_USER,
+                "invalid string",
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL,
+                String.valueOf(true));
+    }
+
+    @Test
+    public void testExecute_withFalseIsFetchingAll_shouldUpdateShownAndReturnUnreadNotifications() {
+        loginAsInstructor(GOOGLE_ID);
+
+        List<Notification> testAllNotifications = new ArrayList<>();
+        List<Notification> testUnreadNotifications = new ArrayList<>();
+        Set<String> readNotificationsId;
+
+        for (int i = 0; i < UNREAD_NOTIFICATION_COUNT; i++) {
+            testUnreadNotifications.add(getTypicalNotificationWithId());
+        }
+
+        for (int i = 0; i < READ_NOTIFICATION_COUNT; i++) {
+            Notification readNotification = getTypicalNotificationWithId();
+            readNotification.setShown();
+            testAllNotifications.add(readNotification);
+        }
+
+        readNotificationsId = testAllNotifications.stream()
+                .map(n -> n.getId().toString())
+                .collect(Collectors.toSet());
+
+        testAllNotifications.addAll(testUnreadNotifications);
+
+        when(mockLogic.getAllNotifications()).thenReturn(testAllNotifications);
+        when(mockLogic.getActiveNotificationsByTargetUser(NotificationTargetUser.INSTRUCTOR))
+                .thenReturn(testUnreadNotifications);
+
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.INSTRUCTOR.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(false),
+        };
+
+        GetNotificationsAction action = getAction(requestParams);
+        JsonResult jsonResult = getJsonResult(action);
+
+        NotificationsData output = (NotificationsData) jsonResult.getOutput();
+        List<NotificationData> notifications = output.getNotifications();
+        verifyDoesNotContainNotifications(notifications, readNotificationsId);
+
+        // should update notification has shown attribute
+        List<Notification> notificationAttributes =
+                mockLogic.getActiveNotificationsByTargetUser(NotificationTargetUser.INSTRUCTOR);
+        notificationAttributes = notificationAttributes.stream()
+                .filter(n -> !readNotificationsId.contains(n.getId().toString()))
+                .collect(Collectors.toList());
+        notificationAttributes.forEach(n -> assertTrue(n.isShown()));
```

Comment:
Let's rename `notificationAttributes` to some other names such as `activeNotifications` becuase 1) `XXXAttribute` is the naming convention for datastore entities 2) it is clearer what kind these notifications are of